hibernate-core/src/main/java/org/hibernate/boot/model/internal/EmbeddableBinder.java

```diff
@@ -451,6 +611,75 @@ private static List<PropertyData> collectClassElements(
 		return classElements;
 	}
 
+	private static void collectSubclassElements(
+			AccessType propertyAccessor,
+			MetadataBuildingContext context,
+			XClass superclass,
+			List<PropertyData> classElements,
+			BasicType<?> discriminatorType,
+			Map<Object, String> discriminatorValues,
+			Map<String, String> subclassToSuperclass) {
+		for ( final XClass subclass : context.getMetadataCollector().getEmbeddableSubclasses( superclass ) ) {
```

Comment:
Is this "safe" i.e. are all embeddable subclasses known at this point for sure?

Reply:
Yes, I believe so, all embeddable subclasses are registered in `org.hibernate.boot.model.internal.AnnotationBinder#buildInheritanceStates`